### PR TITLE
Feat : Design DevOps Section of Contribution page

### PR DESCRIPTION
**Pull request by cp-jagruti-a**
## Review checklist

**Note:** Make sure all points should be checked before submiting PR for review, otherwise PR will be cosidered invalid.

- [x] Follow guidelines for creating branch name, adding commit messages or PR title
- [x] Add MR description if applicable
- [x] Follow guidelines for naming conventions everywhere (i.e files/folders, data structures)
- [x] Reuse code (if the same code is written twice, make it common and reuse it at both places)
- [x] Remove unused or commented code if not required
- [x] Make MR self-approved - review MR as a reviewer and give it self-approval if everything is ok, if not make the required changes
- [x] In progress MR should be marked as draft
- [x] Make MR mark as ready before submitting it for review
- [x] Add images, videos, or gifs of UI 
- [x] Add page speed score screenshot of UI (It should be > 85 for mobile and > 90 for desktops)

**Design should be tested on:**
- [x] Inspector (all iPhones and android devices like samsung, motorola)
- [x] Desktop (chrome, firefox, safari)
- [ ] Laptop / Macbook  (chrome, firefox, safari)
- [x] iPhone(safari)
- [x] Available android devices  (chrome, firefox)
- [x] iPad(safari)
- [x] Tablet (chrome, firefox)

**Ui Video :** 

https://github.com/canopas/canopas-website/assets/85920585/d12e0230-9a22-409c-983b-6c18df6343b3

**Performance:**
**Desktop:**
Screenshot from 2023-06-14 16-03-57
**Mobile**
devops-mobiles

### CODE REVIEW

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        <swiper-slide v-for="(blog, index) in devOps" :key="index">
+          <div
+            class="tw-transition-all tw-duration-200 tw-ease-out tw-scale-[0.75]"
+          >
+            <img
+              @click="openBlog(blog)"
+              :src="blog.image[0]"
+              :srcset="`${blog.image[0]} 400w, ${blog.image[1]} 800w`"
+              class="tw-relative tw-w-full tw-h-full tw-object-cover tw-rounded-[10px]"
+              alt="favourite"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:19]: Replace it with suitable name

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+export default {
+  data() {
+    return {
+      devOps: [
+        {
+          id: 1,
+          image: [devops1_400w, devops1_800w],
+          title:
+            "Deploy a website with AWS S3 + CloudFront using CloudFormation",
+          time: "8 min read",
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:151]: rename it to `readTime`

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+          >
+            <img
+              @click="openBlog(blog)"
+              :src="blog.image[0]"
+              :srcset="`${blog.image[0]} 400w, ${blog.image[1]} 800w`"
+              class="tw-relative tw-w-full tw-h-full tw-object-cover tw-rounded-[10px]"
+              alt="favourite"
+            />
+            <div
+              class="time tw-absolute tw-top-0 tw-right-0 tw-rounded-bl-[5px] tw-bg-white tw-px-3 tw-py-1 tw-font-inter-medium tw-text-[0.875rem] tw-leading-[1.313rem]"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:22]: Is `time` class used?

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        </swiper-slide>
+      </swiper>
+    </div>
+    <div
+      class="tw-container tw-hidden lg:tw-flex tw-flex-col tw-gap-10 tw-mt-16"
+    >
+      <div class="tw-flex tw-flex-row tw-gap-10">
+        <div v-for="blog in devOps.slice(0, 2)" :key="blog.id">
+          <div
+            class="tw-relative tw-w-full tw-h-full tw-group"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:36]: What `group` class will do?

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+            @click="openBlog(blog)"
+          >
+            <img
+              :src="blog.image[0]"
+              :srcset="`${blog.image[0]} 400w, ${blog.image[1]} 800w`"
+              alt="favourite"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:42]: Replace it with suitable name

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+              class="tw-w-full tw-h-full tw-object-cover tw-rounded-[20px] tw-cursor-pointer"
+              loading="lazy"
+            />
+            <div
+              class="gradient-class tw-absolute tw-left-[0] tw-bottom-[0] tw-w-[55%] tw-h-[20%] tw-rounded-b-[10px] tw-opacity-1 tw-transition-all tw-duration-1000 tw-ease group-hover:tw-w-[100%] group-hover:tw-h-[85%]"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:47]: In figma, this div covers almost 80% of the image horizontally. And in our design, it covers around 50% of the image.
Also need to reduce height of div

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            ></div>
+            <div
+              class="tw-absolute tw-bottom-10 tw-p-[1rem] tw-opacity-0 tw-text-white tw-font-inter-medium tw-transition-all tw-duration-1000 tw-translate-y-[50%] group-hover:tw-translate-y-[0%] group-hover:tw-transition-all group-hover:tw-duration-1000 group-hover:tw-opacity-[1]"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:50]: Text and div animations timings are not aligned. They should come on same time.
In current scenario, text are visible earlier then div

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+              <p
+                class="tw-mt-4 tw-w-[75%] tw-text-[1.5rem] tw-leading-[2.25rem]"
+              >
+                {{ blog.title }}
+              </p>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:64]: Add comment for separating mobile and desktop divs

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/components/contributions/DevOps.vue (hunk 6 above)
Comment (cp-jagruti-a) [reply vue-frontend/src/components/contributions/DevOps.vue:47]: you are talking about div before hover or after?

Comment (cp-sumi-k) [reply vue-frontend/src/components/contributions/DevOps.vue:50]: Before hover, it should covers 80% of image
After hover, animation timings should be aligned

File: vue-frontend/src/components/contributions/DevOps.vue [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+          title:
+            "Complete guide to deploying SSR Vite apps on AWS with automation",
+          readTime: "5 min read",
+          link: "https://blog.canopas.com/complete-guide-to-deploying-ssr-vite-apps-on-aws-with-automation-27676113d6ac",
+        },
+      ],
+    };
+  },
+  methods: {
+    openBlog(blog) {
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/DevOps.vue:193]: Pass only link in param

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**